src/freon.c

```diff
@@ -101,6 +101,79 @@ static int set_display_lock(unsigned int pid) {
     return 0;
 }
 
+static int32_t crtc_set_prop(int fd, uint32_t crtc_id,
+			     drmModeObjectPropertiesPtr props,
+			     const char *name, uint64_t value)
+{
+    uint32_t u;
+    int32_t ret;
+    drmModePropertyPtr prop;
+
+    for (u = 0; u < props->count_props; u++) {
+        prop = drmModeGetProperty(fd, props->props[u]);
+	if (!prop)
+	    continue;
+	if (strcmp(prop->name, name)) {
+	    drmModeFreeProperty(prop);
+	    continue;
+	}
+	ret = drmModeObjectSetProperty(fd, crtc_id, DRM_MODE_OBJECT_CRTC,
+                                       props->props[u], value);
+	if (ret < 0)
+	    TRACE("setting property %s failed with %d\n", name, ret);
+	else
+	    ret = 0;
+	drmModeFreeProperty(prop);
+	return ret;
+    }
+    TRACE("could not find property %s\n", name);
+    return -1;
+}
+
+/* Reset CTM/GAMMA properties to avoid artifacts (#3791). */
+static void drm_reset_props()
+{
+    int i, fd;
+    drmModeRes* resources;
+
+    if (!orig_open) preload_init();
+
+    fd = orig_open("/dev/dri/card0", O_RDWR, 0);
+
+    TRACE("%s %d\n", __func__, fd);
+
+    if (fd < 0)
+        return;
+
+    resources = drmModeGetResources(fd);
+    TRACE("%s res=%p\n", __func__, resources);
+    if (!resources)
+        goto closefd;
+
+    for (i = 0; i < resources->count_crtcs; i++) {
+        drmModeObjectPropertiesPtr crtc_props;
+	uint32_t crtc_id = resources->crtcs[i];
+
+	crtc_props = drmModeObjectGetProperties(fd, crtc_id, DRM_MODE_OBJECT_CRTC);
+	TRACE("%s crtc %d %p\n", __func__, i, crtc_props);
+	if (!crtc_props)
+            continue;
+
+        /* Reset color matrix to identity and gamma/degamma LUTs to pass through,
```

Comment:
maybe a reference to where this code is from, in case things change in the future?

Reply:
Looks like you did it yourself, thanks!